File: BFS_OF_GRAPH.java

class Solution {
    // Function to return Breadth First Traversal of given graph.
    public ArrayList<Integer> bfsOfGraph(int v, ArrayList<ArrayList<Integer>> adj) {
        // Code here
        Queue<Integer>q=new LinkedList<>();
        ArrayList<Integer> a=new ArrayList<>();
        int vis[]=new int[v+1];
        q.offer(0);
        vis[0]=1;
        while (!q.isEmpty())
        {
            int n=q.poll();
            a.add(n);
            for(int x:adj.get(n))
            {
                if (vis[x] !=1)
                {
                    q.offer(x);
                    vis[x]=1;
                }
            }
        }
        return a;
    }
}

File: Difficulty: Easy/BFS of graph/bfs-of-graph.py
#User function Template for python3
from collections import deque
class Solution:
    # Function to return Breadth First Search Traversal of given graph.
    def bfs(self, adj):
        # code here
        q=deque()
        v=set()
        q.append(0)
        v.add(0)
        l=[]
        while q :
            x=q.popleft()
            l.append(x)
            for y in adj[x] :
                if y not in v :
                    q.append(y)
                    v.add(y)
        return l
#{ 
 # Driver Code Starts
import sys


#Position this line where user code will be pasted.
def main():
    tc = int(sys.stdin.readline().strip())

    for _ in range(tc):
        V = int(sys.stdin.readline().strip())  # Number of vertices
        adj = []  # Adjacency list

        for _ in range(V):
            input_line = sys.stdin.readline().strip()
            node = list(map(int, input_line.split())) if input_line else []
            adj.append(node)

        obj = Solution()
        ans = obj.bfs(adj)
        print(" ".join(map(str, ans)))
        print("~")


if __name__ == "__main__":
    main()

# } Driver Code Ends